PowerShellGet/PSModule.psm1

```diff
@@ -2873,18 +3005,65 @@ function Publish-Script
+                        if (-not $Force -and ($galleryScriptPrerelease -gt $toPublishScriptPrerelease))
+                        {
+                            $message = $LocalizedData.ScriptVersionShouldBeGreaterThanGalleryVersion -f ($scriptName,
+                                                                                                     $PSScriptInfo.Version,
+                                                                                                     $currentPSGetItemInfo.Version,
+                                                                                                     $currentPSGetItemInfo.RepositorySourceLocation)
+                            ThrowError -ExceptionName "System.InvalidOperationException" `
+                                    -ExceptionMessage $message `
+                                    -ErrorId "ScriptVersionShouldBeGreaterThanGalleryVersion" `
+                                    -CallerPSCmdlet $PSCmdlet `
+                                    -ErrorCategory InvalidOperation
+                        }
+
+                        # elseif ($galleryScriptPrerelease -lt $toPublishScriptPrerelease) --> allow publish
+                    }
+                }
+                elseif (-not $Force -and ($galleryScriptVer -gt $toPublishScriptVer))
                 {
                     $message = $LocalizedData.ScriptVersionShouldBeGreaterThanGalleryVersion -f ($scriptName,
-                                                                                                 $PSScriptInfo.Version,
-                                                                                                 $currentPSGetItemInfo.Version,
-                                                                                                 $currentPSGetItemInfo.RepositorySourceLocation)
+                                                                                                     $PSScriptInfo.Version,
+                                                                                                     $currentPSGetItemInfo.Version,
+                                                                                                     $currentPSGetItemInfo.RepositorySourceLocation)
                     ThrowError -ExceptionName "System.InvalidOperationException" `
-                               -ExceptionMessage $message `
-                               -ErrorId "ScriptVersionShouldBeGreaterThanGalleryVersion" `
-                               -CallerPSCmdlet $PSCmdlet `
-                               -ErrorCategory InvalidOperation
+                            -ExceptionMessage $message `
+                            -ErrorId "ScriptVersionShouldBeGreaterThanGalleryVersion" `
+                            -CallerPSCmdlet $PSCmdlet `
+                            -ErrorCategory InvalidOperation
                 }
+
+                # else ($galleryScriptVer -lt $toPublishScriptVer) --> allow publish

```

Comment:
Sure, I can do that.  These code snippets used to be cases in the if-elseif-else logic, but I realized that no action was needed if they passed,  so I didn't add them.  But I still wanted to show that I had actually considered those cases and what the results of those cases are.

---
In reply to: [132592393](https://github.com/rebro-msft/PowerShellGet/pull/1#discussion_r132592393) [](ancestors = 132592393)